src/test/mochitest/head.js

```diff
@@ -343,13 +344,22 @@ function isPaused(dbg) {
  * @static
  */
 async function waitForPaused(dbg) {
-  // We want to make sure that we get both a real paused event and
-  // that the state is fully populated. The client may do some more
-  // work (call other client methods) before populating the state.
-  let loading = waitForDispatch(dbg, "LOAD_OBJECT_PROPERTIES");
-  await waitForThreadEvents(dbg, "paused");
-  await waitForState(dbg, state => isPaused(dbg));
-  await loading;
+  function scopeIsLoaded(dbg) {
+    const { getLoadedObjects } = dbg.selectors;
+
+    const loadedObjects = Object.values(getLoadedObjects(dbg.getState()))
+    if (loadedObjects.length == 0) {
+      return false;
+    }
+
+    return Object.keys(loadedObjects[0]).length !== 0;;
+  }
+
+  await waitForState(
+    dbg,
+    state => isPaused(dbg) && scopeIsLoaded(dbg),
```

Comment:
Is the `state` being passed to this function the same value retrieved by `dbg.getState()`? It seems like it would make sense for `scopeIsLoaded` to be a predicate on states.

Reply:
👍